libafl_qemu/src/emu/hooks.rs

```diff
@@ -1221,8 +1245,10 @@ where
         self.hooks.syscalls_function(hook)
     }
 
+    /// # Safety
+    /// Calls through to the, potentially unsafe, `syscalls_closure`
     #[allow(clippy::type_complexity)]
-    pub fn syscalls_closure(
+    pub unsafe fn syscalls_closure(
```

Comment:
@rmalmain would you say the closure are actually all safe to call?

Reply:
i'd say yes (at least as of today).
rust implements Box<dyn Fn> with a fat pointer, so the transmute *should* never unsafe.
in practice, i don't think the language guarantees anything about this anywhere, it just happens to work with the current implementation of closures.

Reply:
so should I remove the `unsafe` thing here?

Reply:
hum, do we consider it's safe to assume the compiler will keep this implementation the same internally in the future? no sure what to think there tbh

Reply:
Is there any way to (debug_)assert this?

Reply:
not really imho, it's a semantic thing.
transmute compiles only if both types have the same size, maybe we can consider this enough?

Reply:
So I'll remove the unsafe again and we YOLO it?